signer/src/error.rs

```diff
@@ -11,6 +11,32 @@ use crate::stacks::contracts::WithdrawalAcceptValidationError;
 /// Top-level signer error
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
```

Comment:
Just a random comment, nothing to do here:

This enum feels like it just keeps growing and growing; I'm thinking we maybe should start thinking about breaking it up. Like, we could have a `BitcoinCore(BitcoinCoreError)` variant which would break out at least a handful of these errors, for example.

Reply:
Not a bad idea, it's quite common to follow such an approach. I'm on board with doing so. 

The main thing that I want to encourage is to preserve context in our error variants. And we can do that by having dedicated error types that each capture enough context. Most of our error variants fall short of capturing enough context, but still that's the aspiration. Also, implementing `From` on the "main" error type keeps things ergonomic so it's all good.

Note that a little history on error handling in the signer crate can be found in https://github.com/stacks-network/sbtc/pull/161#discussion_r1604451042 and https://github.com/stacks-network/sbtc/issues/229.